tests/api_test.py

```diff
@@ -1437,6 +1439,7 @@ def test_hessian(self):
     assert np.allclose(hessian(f)(x), A + A.T)
 
   @jtu.skip_on_devices("tpu")
+  @jax.numpy_dtype_promotion('standard')  # Test explicitly exercises implicit dtype promotion.
```

Comment:
Same everywhere.

Reply:
In this case, since the holomorphic hessian specifically requires certain types for its inputs, we would lose test coverage if we only tested the case where all inputs are complex. It seems better to me to keep the current test coverage rather than reducing its scope.